### ISSUE TITLE
Reading the Bitcoin database causes an error about the levelDB mismatching

### ISSUE THREAD

**Issue by ChrisTX** [Congyuwang/Rusty-Bitcoin-Explorer#29]
Hi,
I'm trying to read the Bitcoin transaction history for an analysis project. Doing so worked fine but as of a few weeks ago causes the following error:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `788806`,
 right: `788805`: some block info missing from block index levelDB,delete Bitcoin folder and re-download!', /home/cpfeiffer/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoin-explorer-1.2.19/src/parser/block_index.rs:66:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
I have redownloaded the database _twice_ from scratch to no avail, it still causes the error, and always with the same amount of blocks mismatching. As of such, I suspect it's due to a recent anomaly in Bitcoin rather than corrupted data (recent because the exact same code used to work fine until a while ago).

**Comment by ChrisTX** [comment]
Unfortunately it did not. It still emits the same assertion failure.

**Comment by ChrisTX** [comment]
This changed the error to the following:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `1`: some block info missing from block index levelDB,delete Bitcoin folder and re-download!', /home/cpfeiffer/Rusty-Bitcoin-Explorer/src/parser/block_index.rs:66:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by Congyuwang** [comment]
#30

**Comment by ChrisTX** [comment]
Interestingly enough, it fails at a different block now:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `788838`,
 right: `788837`: some block info missing from block index levelDB,delete Bitcoin folder and re-download!', /home/cpfeiffer/Rusty-Bitcoin-Explorer/src/parser/block_index.rs:66:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
I also let it print out the details using your prior commit:
```
788836, BlockIndexRecord { version: 220000, height: 788835, status: 157, n_tx: 3965, n_file: 3585, n_data_pos: 47318060, header: BlockHeader { version: 629039104, prev_blockhash: 00000000000000000002ce9a66b3b635737634ea3ed4b7b366f6bd9a3b6303d7, merkle_root: cd3e69cdac30ed4781081502296ec8af104c58773553d3459cbc766e02a5da24, time: 1683580590, bits: 386260225, nonce: 2161909841 } }
788837, BlockIndexRecord { version: 220000, height: 788836, status: 157, n_tx: 4522, n_file: 3585, n_data_pos: 49068596, header: BlockHeader { version: 536903680, prev_blockhash: 0000000000000000000343770ef3abc9b03a01a1c4466cfe64479f180ccbb971, merkle_root: 789c8c9303502c1a3aef4b1b9beb329097ab0b45ec5369c98bcef2a846d0e57c, time: 1683581626, bits: 386260225, nonce: 2342489441 } }
788838, BlockIndexRecord { version: 220000, height: 788837, status: 157, n_tx: 4485, n_file: 3585, n_data_pos: 52457103, header: BlockHeader { version: 743284736, prev_blockhash: 0000000000000000000018e509a36935f22951d9da5ca7b2623ca817b971a09a, merkle_root: bb9bfad983a5c3ce9ebdcb15911f0db01cc5ab69d903d58ccaf4bfb140e4483f, time: 1683582076, bits: 386260225, nonce: 1834508594 } }
788839, BlockIndexRecord { version: 220000, height: 788837, status: 157, n_tx: 4446, n_file: 3585, n_data_pos: 50792339, header: BlockHeader { version: 830185472, prev_blockhash: 0000000000000000000018e509a36935f22951d9da5ca7b2623ca817b971a09a, merkle_root: b92ab690c0ec3dddd8d2a8fba0fc9dea648b2009cd6a6342ba5f3575c1ceec38, time: 1683582103, bits: 386260225, nonce: 1761452616 } }
```

**Comment by Congyuwang** [comment]
Looks like this is because we’ve got two conflicting record for block 788837 🤔

**Comment by Congyuwang** [comment]
Got to get the chain head and walking back to the genesis block.

**Comment by Congyuwang** [comment]
Thank you for your feedback. I've updated the branch, and it has passed the test for the first 10000 blocks or so based on my test data. Can you test it again?

**Comment by ChrisTX** [comment]
The current version seems to run fine without any issues. Thanks a lot!

**Comment by Congyuwang** [comment]
Released v1.2.20
